augly/image/functional.py

```diff
@@ -1786,3 +1786,64 @@ def vflip(
     imutils.get_metadata(metadata=metadata, function_name="vflip", **func_kwargs)
 
     return imutils.ret_and_save_image(aug_image, output_path)
+
+
+def skew(
+    image: Union[str, Image.Image],
+    output_path: Optional[str] = None,
+    metadata: Optional[List[Dict[str, Any]]] = None,
+    level: float, 
+    max_level,
+    axis: str
+) -> Image.Image:
+    
+    """
+    Skew image with respect to X-axis or Y-axis
+    
+    @param image: the path to an image or a variable of type PIL.Image.Image
+        to be augmented
+        
+    @param level: Level of the operation that will be in [0, `PARAMETER_MAX`]
+    
+    @param maxval: Maximum value that the operation can have. This will be 
+                scaled to level/PARAMETER_MAX.
+                
+    @param output_path: the path in which the resulting image will be stored.
+        If None, the resulting PIL Image will still be returned
+
+    @param metadata: if set to be a list, metadata about the function execution
+        including its name, the source & dest width, height, etc. will be appended
+        to the inputted list. If set to None, no metadata will be appended or returned
+
+    @returns: the augmented PIL Image
+    """
+    func_kwargs = imutils.get_func_kwargs(metadata, locals())
+    image = imutils.validate_and_load_image(image)
+    if axis == 'x-axis':
```

Comment:
You have a lot of duplicate code in both cases in this if block. Please move the code that is the same outside of the if. In fact, it looks like only the `data` arg to `image.transform()` depends on the axis, so we can do everything in the same code block & just assign the `data` arg depending on the value of `axis`, like so:
```python
            level = float(np.random.uniform(low=0.1, high=level)) * max_level / 10
            w, h = image.size

            if np.random.uniform() > 0.5:
                level = -level

            data = (1, level, 0, 0, 1, 0) if axis == "x-axis" else (1, 0, 0, level, 1, 0)
            aug_img = image.transform(
                (w, h), Image.AFFINE, data, resample=Image.BILINEAR
            )
            imutils.get_metadata(
                metadata=metadata, function_name="skew", aug_img=aug_img, **func_kwargs
            )
            return aug_img
```

Reply:
@tanujdhiman please make this change, or justify why not.

Reply:
@tanujdhiman again, please make this change or explain why not.

Reply:
Sorry, I can't get your point. :(